ksqldb-parser/src/main/java/io/confluent/ksql/parser/properties/with/CreateSourceProperties.java

```diff
@@ -182,6 +238,14 @@ public CreateSourceProperties withPartitionsAndReplicas(
     return new CreateSourceProperties(originals, durationParser);
   }
 
+  public CreateSourceProperties withFormats(final String keyFormat, final String valueFormat) {
+    final Map<String, Literal> originals = props.copyOfOriginalLiterals();
+    originals.put(CommonCreateConfigs.KEY_FORMAT_PROPERTY, new StringLiteral(keyFormat));
+    originals.put(CommonCreateConfigs.VALUE_FORMAT_PROPERTY, new StringLiteral(valueFormat));
+
+    return new CreateSourceProperties(originals, durationParser);
+  }
```

Comment:
There are already QTTs for this. Are there specific cases you'd additionally like to see?

The injector is a no-op if `FORMAT` is supplied, since the injector returns if both a key and value format are present. As a consequence, this method is never called if `FORMAT` is supplied. If it feels more "correct" to clear `FORMAT` in this method I can do that but it's not necessary the way the code is set up today.